packages/dev/docs/pages/blog/building-a-combobox.mdx

```diff
@@ -0,0 +1,129 @@
+it is portalled to the end of the document, meaning there could be any number of elements that a screenreader user would have to navigate through just to move from the ComboBox input to the menu itself. For touch screen reader users, it would be particularly problematic
+since they may not have access to a physical keyboard and thus can't use the arrow keys as to navigate through menu items.
+
+This put us in a bind because removing the portal strategy wasn't really an option. If we didn't portal the popover menu out to the end of the document, we'd run into the risk of the combobox's menu being cut off if the combobox's parent has `overflow: hidden` or `overflow: scroll`
+set. Therefore, we had to figure out a way to make the ComboBox input and the ComboBox menu the only elements accessible to screen readers while the menu is open so that a screen reader user wouldn't accidentally leave the ComboBox context.
+
+The solution that we came up with was to crawl the DOM and apply `aria-hidden` to every element that wasn't the ComboBox input or menu. To crawl the DOM we used a [TreeWalker](https://developer.mozilla.org/en-US/docs/Web/API/TreeWalker),
+setting up a node filter to determine if a node should be left visible, hidden, or skipped in the case where its parent was already hidden. In addition, we watch for any changes in the DOM while the menu is open
+via a [MutationObserver](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver), hiding those new elements if need be. When the menu closes, every node that we modified is reverted back to its previous state.
+Check out our `ariaHideOutside` hook in `@react-aria/overlays` if you'd like to learn more.
+
+### Mobile implementation divergence
+
+Another divergence between the WAI-ARIA 1.2 ComboBox spec and our ComboBox implementation came as a result of our mobile implementation. Initially, we considered the outer input and the tray input as both being individual combo boxes that mirror each others state. This became increasingly complicated as we tried to
+manage things like focus and internal state between the two inputs, resulting in several tray input specific affordances added to our `useComboBox` hook, reducing the aria hook's generality by introducing this React Spectrum specific behavior. Interacting with the tray input with a screen reader was also problematic
+since the screen reader would detect that the input had `role="combobox"` and thus erroneously state "double tap to close", a touch action usually reserved for focusing an input. After some deliberation, we decided to alter the accessibility implementation for the mobile experience, removing the outer combo box input in favor for a button that
+looks like a combo box. Tapping the button opens the tray that contains the actual combo box input, albeit with `role="searchbox"` instead of `role="combobox"` to avoid the aforementioned screen reader announcement.
```

Comment:
```suggestion
resembled a combo box. Tapping the button opens the tray that contains the actual combo box input, albeit with `role="searchbox"` instead of `role="combobox"` to avoid the aforementioned screen reader announcement.
```